velox/functions/lib/DateTimeFormatter.h

```diff
@@ -175,10 +175,22 @@ class DateTimeFormatter {
 
   DateTimeResult parse(const std::string_view& input) const;
 
+  /// Returns max size of the formatted string. Can be used to preallocate
+  /// memory before calling format() to avoid extra copy.
+  uint32_t maxResultSize(const date::time_zone* timezone) const;
```

Comment:
Would be nice to add some unit tests.

Reply:
Just added some test cases. And the existing tests for DateTimeFormatter, format_datetime & date_format also cover the verification of the new API.